core/src/executor/context/blend_context.rs

```diff
@@ -33,49 +33,41 @@ impl<'a> BlendContext<'a> {
     }
 
     pub fn get_value(&'a self, target: &str) -> Option<&'a Value> {
-        let get_value = || {
-            self.columns
-                .iter()
-                .position(|column| column == target)
-                .map(|index| match &self.row {
-                    BlendContextRow::Shared(row) => row.get_value(index),
-                    BlendContextRow::Single(Some(row)) => row.get_value(index),
-                    BlendContextRow::Single(None) => Some(&Value::Null),
-                })
+        let value = match &self.row {
+            BlendContextRow::Shared(row) => row.get_value(&self.columns, target),
+            BlendContextRow::Single(Some(row)) => row.get_value(&self.columns, target),
+            BlendContextRow::Single(None) => Some(&Value::Null),
         };
 
-        match get_value() {
-            None => match &self.next {
-                None => None,
-                Some(context) => context.get_value(target),
-            },
-            Some(value) => value,
+        if value.is_some() {
+            return value;
         }
+
+        self.next
+            .as_ref()
+            .and_then(|context| context.get_value(target))
     }
 
     pub fn get_alias_value(&'a self, table_alias: &str, target: &str) -> Option<&'a Value> {
-        let get_value = || {
+        let value = (|| {
             if self.table_alias != table_alias {
                 return None;
             }
 
-            self.columns
-                .iter()
-                .position(|column| column == target)
-                .map(|index| match &self.row {
-                    BlendContextRow::Shared(row) => row.get_value(index),
-                    BlendContextRow::Single(Some(row)) => row.get_value(index),
-                    BlendContextRow::Single(None) => Some(&Value::Null),
-                })
-        };
+            match &self.row {
+                BlendContextRow::Shared(row) => row.get_value(&self.columns, target),
+                BlendContextRow::Single(Some(row)) => row.get_value(&self.columns, target),
+                BlendContextRow::Single(None) => Some(&Value::Null),
+            }
+        })();
```

Comment:
```rust
        let value = {
            if self.table_alias != table_alias {
                return None;
            }

            match &self.row {
                BlendContextRow::Shared(row) => row.get_value(&self.columns, target),
                BlendContextRow::Single(Some(row)) => row.get_value(&self.columns, target),
                BlendContextRow::Single(None) => Some(&Value::Null),
            }
        };
```

Reply:
There seems to be no need to use a closure (late review).

Reply:
This closure is required because that causes different behavior.
closure version does not break the flow though `self.table != table_alias` happens but using block statement, it will call `return` and the rest codes will be skipped.

Reply:
because of the next codes
<img width="454" alt="image" src="https://user-images.githubusercontent.com/2025065/195359019-41aa86f6-55d6-46d2-bce4-f4983f63ee55.png">

Reply:
Oh sure.. That's right. Because there is a return statement in the block, it skips.
Looks like I missed it, thanks